test/main.test.js

```diff
@@ -129,4 +130,50 @@ describe('LoopBack Context', function() {
       ], done);
     });
   });
+  it('doesn\'t mix up contexts if using concurrently then() from when 3.7.7',
+  function() {
+    expect(require('when/package.json').version).to.equal('3.7.7');
```

Comment:
I am little bit worried about this long-distance coupling between the tests and package.json, and also what will happen when we will want to test two different versions of a same package.

Could we run `npm install when@3.7.7` from the test instead? To speed up local `npm test` and prevent repeated fetching of packages from the registry, we can use https://www.npmjs.com/package/strong-cached-install to cache the results - I have had good experience using this approach in other LoopBack projects. We will need to expose API for installing a single package, but that should be trivial - see the existing [installPackage](https://github.com/strongloop/strong-cached-install/blob/91e7e428150d32749bbb1e066f127f42935d3d70/index.js#L51) function.